src/KafkaFlow.Admin/MemoryTelemetryStorage.cs

```diff
@@ -60,11 +60,11 @@ private void TryCleanItems()
 
         private void CleanExpiredItems()
         {
-            foreach (var metric in this.metrics.ToList())
+            foreach (var metricKey in this.metrics.Keys)
```

Comment:
I think you can deconstruct the key and value here and avoid the `this.metrics[metricKey]` at line 65.